docs/site/migration/models/remoting-hooks.md

```diff
@@ -6,8 +6,531 @@ sidebar: lb4_sidebar
+
+Hooks can be configured to run for one or more methods using a wildcard
+specifier. In general, there are three kinds of hook scopes:
+
+- **Global hooks** are executed for _every_ remote method.
+
+  These hooks are registered on the application object using a `**` wildcard.
+  For example:
+
+  ```js
+  app.beforeRemote('**', handlerFn);
+  ```
+
+- **Model level hooks** are executed for every remote method of a given model
+  class.
+
+  These hooks are typically registered on the application object using
+  `{model-name}.**` wildcard. For example, to run a hook for all methods of the
+  `User` model:
+
+  ```js
+  app.beforeRemote('User.**', handlerFn);
+  ```
+
+- **Method level hooks** are executed only for a single method of a given model
+  class.
+
+  These hooks are typically registered on the model class. For example, to run a
+  hook whenever `User.login` is called remotely:
+
+  ```js
+  User.beforeRemote('login', handlerFn);
+  ```
+
+LoopBack 4 provides [Interceptors](../../Interceptors.md) feature to enable
+application developers to implement similar functionality.
+
+- [**Global interceptors**](../../Interceptors.html#global-interceptors) are
+  executed for _every_ request handled by a LoopBack 4 controller method or a
+  LoopBack 4 route handler. They correspond to LoopBack 3 **global hooks**.
+- [**Class level interceptors**](../../Interceptors.html#class-level-interceptors)
+  are executed for requests handled by the given
+  [Controller](../../Controllers.html) class. They correspond to LoopBack 3
+  **model level hooks**.
+- [**Method level interceptors**](../../Interceptors.md#method-level-interceptors)
+  are executed only for request handled by the given controller method. They
+  correspond to LoopBack 3 **method level hooks**.
+
+{% include note.html content=" In LoopBack 3, a model class has three
+responsibilities: a model describing shape of data, a repository providing
+data-access APIs, and a controller implementing REST API). Both remoting hooks
+(invoked by REST layer) and operation hooks (invoked by ORM/data-access layer)
+are registered for models.
+
+In LoopBack 4, the REST API is implemented by controllers that are decoupled
+from models, therefore interceptor are registered and invoked on controller
+classes/method, not on a model classes/methods. " %}
+
+In the following sections, we will first explain how to rewrite a LoopBack 3
+hook implementation to a LoopBack 4 interceptor implementation. Then we show how
+to create interceptors replacing LoopBack 3 remoting hooks registered at global,
+model (class) level and method level.
+
+## Rewriting a LoopBack 3 hook to a LoopBack 4 interceptor
+
+Let's take a look at a typical implementation of a LoopBack 4 interceptor:
+
+```ts
+async function intercept(
+  invocationCtx: InvocationContext,
+  next: () => ValueOrPromise<InvocationResult>,
+) {
+  try {
+    // Add pre-invocation logic here
+    const result = await next();
```

Comment:
Just a suggestion.. Referencing the next paragraph, would it be helpful to mention as a comment here something like:
```
// add code from beforeRemote hook here
```

If that's the case, we can also flip the order of the 2 paragraphs too.

Reply:
This code snippet is showing the the interceptor implementation as created by `lb4 interceptor`, I'd like to keep it as-is.

I like the idea of making it more explicit where to put before/after remote hooks, I'll update the code snippet in the example below.